Source/Core/InputCommon/GCAdapter.cpp

```diff
@@ -83,6 +88,22 @@ static void Read()
   }
 }
 
+static void Write()
+{
+  int size = 0;
+  while (s_adapter_thread_running.IsSet())
+  {
+    if (s_rumble_data_available.WaitFor(std::chrono::milliseconds(100)))
```

Comment:
Is there a reason you're using WaitFor instead of just Wait? This will introduce a random 0-100ms delay while shutting down.